src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -52,13 +54,32 @@ private IProjectThreadingService ProjectThreadingService
 
         public DebugPageViewModel()
         {
+            // Hook into our own property changed event. This is solely to know when an active profile has been edited
+            PropertyChanged += ViewModel_PropertyChanged;
         }
 
+
         // for unit testing
-        internal DebugPageViewModel(bool useTaskFactory, UnconfiguredProject unconfiguredProject)
+        internal DebugPageViewModel(TaskCompletionSource<bool> snapshotComplete, UnconfiguredProject unconfiguredProject)
         {
-            _useTaskFactory = useTaskFactory;
+            _firstSnapshotCompleteSource = snapshotComplete;
             UnconfiguredProject = unconfiguredProject;
+            PropertyChanged += ViewModel_PropertyChanged;
+        }
+
+        /// <summary>
+        /// This is here so that we can clear the in-memory status of teh active profile if it has been edited. This is
+        /// so that the profile, and hence the users customizations, will be saved to disk
+        /// </summary>
+        private void ViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (!IgnoreEvents)
+            {
+                if(SelectedDebugProfile != null && SelectedDebugProfile.IsInMemoryObject() && SelectedDebugProfile is IWritablePersistOption writablePersist)
```

Comment:
>IsInMemoryObject [](start = 72, length = 16)

why do we need to check if it is in-memory here?

Reply:
We could just blindly set the property to false. It was more to make it clearer why this code is here.